plutus-core/executables/debugger/FileBrowser.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Logic for the file browsing stage.
+module FileBrowser where
+
+import Common
+import Types
+
+import Brick.Types qualified as B
+import Brick.Widgets.Border qualified as BB
+import Brick.Widgets.Center qualified as BC
+import Brick.Widgets.Core qualified as B
+import Brick.Widgets.Edit qualified as BE
+import Brick.Widgets.FileBrowser qualified as BF
+import Control.Monad.State
+import Data.Text.IO qualified as Text
+import Graphics.Vty qualified as Vty
+import Lens.Micro
+
+drawFileBrowser ::
+    BF.FileBrowser ResourceName -> [B.Widget ResourceName]
+drawFileBrowser fb = [BC.center ui]
+  where
+    ui =
+        BC.hCenter
+            . B.vLimit 40
+            . B.hLimit 120
+            . BB.borderWithLabel (B.txt "Choose a UPLC program file")
+            $ B.vBox [BF.renderFileBrowser True fb, footer]
+
+-- | Handle events in the file browsing stage
+handleFileBrowserEvent :: Vty.Event -> B.EventM ResourceName DebuggerState ()
+handleFileBrowserEvent ev = do
+    B.zoom dsFileBrowser $ BF.handleFileBrowserEvent ev
+    case ev of
+        Vty.EvKey Vty.KEnter [] -> do
+            fb <- gets (^. dsFileBrowser)
+            case BF.fileBrowserSelection fb of
+                [] -> pure ()
+                debuggee : _ -> do
+                    -- A debuggee is selected. Entering the Debugging stage.
+                    uplcText <-
+                        liftIO $
+                            Text.readFile (BF.fileInfoFilePath debuggee)
+                    modify' $ \st' ->
+                        st'
+                            & dsAppStage .~ Debugging
+                            & dsUplcEditor .~ BE.editorText EditorUplc Nothing uplcText
+        _ -> pure ()
+    pure ()
+
+fileBrowserKeyBindings :: forall n. B.Widget n
+fileBrowserKeyBindings =
+    B.vBox
+        [ B.withAttr menuAttr . B.vBox $
+            [ B.txt "Move cursor                      (Up/Down)"
```

Comment:
Where do these come from? Is this just a listing of the standard key bindings in the standard brick file browser widget?

Reply:
Also I'm slightly surprised that upstream doesn't provide something like this, maybe you should make an issue?

Reply:
> I'm slightly surprised that upstream doesn't provide something like this

I guess there are lots of ways to show key bindings, and this is just one of them, so it doesn't make sense for the upstream to provide it.